### DISCUSSION TITLE
`TSRegressor` vs `TSForecaster`?

### DISCUSSION THREAD

**Discussion by deven-gqc** [timeseriesAI/tsai#652] [Q&A]
Hello people,

I have been using tsai at my workplace which has yielded in very promising results. I had a few questions about the entire API.

1. What is the difference between [TSRegressor 1](https://timeseriesai.github.io/tsai/tslearner.html#TSRegressor) and [TSForecaster](https://timeseriesai.github.io/tsai/tslearner.html#TSForecaster)? Their source looks quite similar to me.
2. In order to predict say n timesteps in the future, I have to write a custom function which would make a prediction, then I would use that prediction to make the next prediction and so on. Is there a direct function in the API that would allow that? (darts has such options in the API which allow the same functionality)

Also, [@oguiza](https://forums.fast.ai/u/oguiza) thanks for creating tsai!!

**Comment by oguiza** [comment]
Hi @deven-gqc,
Sorry for my later reply.
TSRegressor and TSForecaster may be similar now but I envision they'll probably evolve in different ways. For example, TSForecaster might include things like fcst_history or fcst_horizon, which would not be applicable to TSRegressor. That's why I separated them.
As to your second question, you may want to take a look at the recent [PatchTST tutorial](https://github.com/timeseriesAI/tsai/blob/main/tutorial_nbs/15_PatchTST_a_new_transformer_for_LTSF.ipynb) that shows how to train a multi-horizon, multi-variate forecasting model. You'll see there's no need to create a custom function.
